debugger/src/main/scala/org.apache.daffodil.debugger.dap/Parse.scala

```diff
@@ -104,16 +106,31 @@ object Parse {
               case "json" => new JsonInfosetOutputter(os, true)
             }
 
+            val dataInputStream = new InputSourceDataInputStream(data)
+            val parseResult = dp.parse(dataInputStream, infosetOutputter)
+            val loc = parseResult.location()
+
             val parse =
               IO.interruptibleMany(
-                dp.parse(
-                  new InputSourceDataInputStream(data),
-                  infosetOutputter
-                )
+                parseResult
                 // WARNING: parse doesn't close the OutputStream, so closed below
               ).ensureOr(res => new Parse.Exception(res.getDiagnostics.toList))(res => !res.isError)
                 .guarantee(IO(os.close) *> done.set(true))
                 .void
+                .flatMap { _ =>
+                  val bitsRead = loc.bitPos1b - 1
+                  val leftOverBits = (dataContents.length - (loc.bytePos1b - 1)) * 8
+
+                  if (bitsRead > 0 && leftOverBits > 0) {
```

Comment:
Why the `bitsRead > 0` condition? It's legal for a DFDL schema to successfully parse but not consume any data, so that bitsRead will be zero. If were was data to consume it this case, it feels like we would still want the left over data message.